collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2043,54 +1948,32 @@ public void onClick(DialogInterface dialog, int id) {
 
     // Cleanup when user exits a form without saving
     private void removeTempInstance() {
-        FormController formController = Collect.getInstance().getFormController();
+        FormController formController = getFormController();
 
         if (formController != null && formController.getInstancePath() != null) {
             SaveToDiskTask.removeSavepointFiles(formController.getInstancePath().getName());
         }
 
-        boolean erase;
-
-        Cursor c = null;
-        try {
-            c = new InstancesDao().getInstancesCursorForFilePath(formController.getInstancePath()
-                    .getAbsolutePath());
-            erase = c.getCount() < 1;
-        } finally {
-            if (c != null) {
-                c.close();
-            }
-        }
-
         // if it's not already saved, erase everything
-        if (erase) {
+        if (!InstancesDaoHelper.isInstanceAvailable(getInstancePath())) {
             // delete media first
-            String instanceFolder = formController.getInstancePath()
-                    .getParent();
+            String instanceFolder = formController.getInstancePath().getParent();
             Timber.i("Attempting to delete: %s", instanceFolder);
-            int images = MediaUtils
-                    .deleteImagesInFolderFromMediaProvider(formController
-                            .getInstancePath().getParentFile());
-            int audio = MediaUtils
-                    .deleteAudioInFolderFromMediaProvider(formController
-                            .getInstancePath().getParentFile());
-            int video = MediaUtils
-                    .deleteVideoInFolderFromMediaProvider(formController
-                            .getInstancePath().getParentFile());
+            File file = formController.getInstancePath().getParentFile();
+            int images = MediaUtils.deleteImagesInFolderFromMediaProvider(file);
+            int audio = MediaUtils.deleteAudioInFolderFromMediaProvider(file);
+            int video = MediaUtils.deleteVideoInFolderFromMediaProvider(file);
 
             Timber.i("Removed from content providers: %d image files, %d audio files and %d audio files.",
                     images, audio, video);
-            File f = new File(instanceFolder);
-            if (f.exists() && f.isDirectory()) {
-                for (File del : f.listFiles()) {
-                    Timber.i("Deleting file: %s", del.getAbsolutePath());
-                    del.delete();
-                }
-                f.delete();
-            }
+            FileUtils.purgeMediaPath(instanceFolder);
         }
     }
 
+    private String getInstancePath() {
```

Comment:
I am confused by this method name because it returns “absolute path” from “instance path”, not “instance path”.  Maybe `getAbsoluteInstancePath` would be better.